DigitalLearningSolutions.Data.Tests/Services/CourseAdminFieldsServiceTests.cs

```diff
@@ -186,5 +194,115 @@ public void RemoveCustomPromptFromCourse_calls_data_service_with_correct_values(
             A.CallTo(() => courseAdminFieldsDataService.DeleteAllAnswersForCourseAdminField(1, 1))
                 .MustHaveHappened();
         }
+
+        [Test]
+        public void GetCustomPromptsWithAnswerCountsForCourse_returns_empty_list_with_no_admin_fields()
+        {
+            // Given
+            const int customisationId = 1;
+            const int centreId = 1;
+            A.CallTo(() => courseAdminFieldsDataService.GetCourseAdminFields(customisationId))
+                .Returns(new CourseAdminFieldsResult());
+
+            // When
+            var result = courseAdminFieldsService.GetCustomPromptsWithAnswerCountsForCourse(customisationId, centreId);
+
+            // Then
+            result.Should().BeEmpty();
+        }
+
+        [Test]
+        public void GetCustomPromptsWithAnswerCountsForCourse_counts_free_text_admin_fields_correctly()
+        {
+            // Given
+            const int customisationId = 1;
+            const int centreId = 1;
+            const int totalDelegatesCount = 10;
+            const int numberOfDelegatesWithAnswer = 3;
+            A.CallTo(() => courseAdminFieldsDataService.GetCourseAdminFields(customisationId))
+                .Returns(
+                    CustomPromptsTestHelper.GetDefaultCourseAdminFieldsResult(
+                        "System Access Granted",
+                        null,
+                        null
+                    )
+                );
+
+            var delegateAnswers = Builder<DelegateCourseAdminFieldAnswers>.CreateListOfSize(totalDelegatesCount)
+                .TheFirst(numberOfDelegatesWithAnswer)
+                .With(a => a.Answer1 = "Answer")
+                .TheRest()
+                .With(a => a.Answer1 = null)
+                .Build();
+            A.CallTo(() => courseDataService.GetDelegateAnswersForCourseAdminFields(customisationId, centreId))
+                .Returns(delegateAnswers);
+
+            // When
+            var result = courseAdminFieldsService.GetCustomPromptsWithAnswerCountsForCourse(customisationId, centreId)
+                .ToList();
+
+            // Then
+            using (new AssertionScope())
+            {
+                result.Should().HaveCount(1);
+                result.First().ResponseCounts.Should()

```

Comment:
Given we know exactly what is expected we could be a bit more exact here and write:
```
result.First().ResponseCounts.Should()
    .BeEquivalentTo(new List<ResponseCounts>
    {
        new ResponseCounts("not blank", numberOfDelegatesWithAnswer),
        new ResponseCounts("blank", totalDelegatesCount - numberOfDelegatesWithAnswer),
    });
```
I think it's a little easier to read too.